Assignment4/sofiaferben/task07.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1FH_-DJcfynrnNfTPZqiwd1u4SkhtbGa8
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""First let's read the RDF file"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+ns = Namespace("http://somewhere#")
+##1) RDFLib
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+  print("Subclass of LivingThing:", s)
+
+##2) SPARQL
+from rdflib.plugins.sparql import prepareQuery
+q1 = prepareQuery('''
+SELECT ?sub
+WHERE {
+  ?sub rdfs:subClassOf ns:LivingThing.
```

Comment:
this will not retrieve the subclasses of subclasses of Living Thing. Hint: use `*`